### PR TITLE
rbac: hide roles on profile page on dotcom

### PR DESCRIPTION
**Pull request by BolajiOlajide**
This PR hides the roles section on the User profile page when a user is accessing the profile page on Dotcom (for both unauthenticated and authenticated users).
It also allows user to view other's roles on customer instances.

CleanShot 2023-03-10 at 21 47 01@2x

## Test plan

* Manual testing
* Update unit tests

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.08 kb) | 0.00% (+0.08 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/bo/hide-roles-on-dotcom/compare-report.html;tab=live_object) for a full comparison between the commits ce75575 and 7a3c007 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by limitedmage — APPROVED** [review]
Nice!

File: client/web/src/user/area/UserArea.tsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ export const UserAreaGQLFragment = gql`
             email
             isPrimary
         }
-        roles {
```
Comment (courier-new) [inline client/web/src/user/area/UserArea.tsx:50]: Ooh fancyyy! Thanks for looking into this, Bolaji. 🙂

File: client/web/src/user/area/UserArea.tsx [hunk 2]
```diff
@@ -47,7 +47,7 @@ export const UserAreaGQLFragment = gql`
...
+        roles @skip(if: $isSourcegraphDotCom) {
```
Comment (courier-new) [inline client/web/src/user/area/UserArea.tsx:50]: Ooh fancyyy! Thanks for looking into this, Bolaji. 🙂

**Pull request merged**